### PR TITLE
ArrayPool fixes

### PR DESCRIPTION
**Pull request by jkotas**
- Use Array.Empty<T>() to make the code identical to CoreRT copy
- Remove unnecessary helper type for lazy initialization
- Use | for clarity

### CODE REVIEW

**Comment by jkotas** [comment]
@stephentoub PTLA

File: src/mscorlib/corefx/System/Buffers/ArrayPool.cs [hunk 1]
```diff
@@ -19,6 +19,16 @@ namespace System.Buffers
     /// </remarks>
     public abstract class ArrayPool<T>
     {
+        /// <summary>Per-type cached pool.</summary>
+        /// <remarks>
+        /// byte[] and char[] are the most commonly pooled array types. For these we use a special pool type
+        /// optimized for very fast access speeds, at the expense of more memory consumption.
+        /// The shared instance is created lazily on first access by the runtime.
+        /// </remarks>
+        private readonly static ArrayPool<T> s_sharedInstance =
```
Comment (justinvp) [inline src/mscorlib/corefx/System/Buffers/ArrayPool.cs]: Nit: `static readonly` while you're at it?

Comment (jkotas) [inline src/mscorlib/corefx/System/Buffers/ArrayPool.cs]: Fixed

File: src/mscorlib/corefx/System/Buffers/ArrayPool.cs [hunk 2]
```diff
@@ -30,25 +40,7 @@ public abstract class ArrayPool<T>
...
-            /// <summary>Per-type cached pool.</summary>
-            /// <remarks>
-            /// byte[] and char[] are the most commonly pooled array types. For these we use a special pool type
-            /// optimized for very fast access speeds, at the expense of more memory consumption.
-            /// </remarks>
-            internal readonly static ArrayPool<T> Value =
-                typeof(T) == typeof(byte) || typeof(T) == typeof(char) ? new TlsOverPerCoreLockedStacksArrayPool<T>() :
-                Create();
-        }
+        public static ArrayPool<T> Shared => s_sharedInstance;
```
Comment (stephentoub) [inline src/mscorlib/corefx/System/Buffers/ArrayPool.cs]: Nit: What you have is fine, but you could also use an auto-prop initializer, e.g.
```C#
public static ArrayPool<T> Shared { get; } =
    typeof(T) == typeof(byte) || typeof(T) == typeof(char) ? new TlsOverPerCoreLockedStacksArrayPool<T>() :
    Create();
```

**Pull request merged**